consensus/blocknode.go

```diff
@@ -58,143 +41,103 @@ type blockNode struct {
 	delayedSiacoinOutputs map[SiacoinOutputID]SiacoinOutput
 }
 
-// CheckTarget returns true if the block id is lower than the target.
-func (b Block) CheckTarget(target Target) bool {
-	blockHash := b.ID()
-	return bytes.Compare(target[:], blockHash[:]) >= 0
+// childDepth returns the depth of a blockNode's child nodes. The depth is the
+// "sum" of the current depth and current target, where sum is defined as:
+//
+//     sum(x,y) := 1/(1/x + 1/y)
+func (bn *blockNode) childDepth() (depth Target) {
+	cumulativeDifficulty := new(big.Rat).Add(bn.target.Inverse(), bn.depth.Inverse())
+	return RatToTarget(new(big.Rat).Inv(cumulativeDifficulty))
 }
 
-// Int returns a Target as a big.Int.
-func (t Target) Int() *big.Int {
-	return new(big.Int).SetBytes(t[:])
+// heavierThan returns true if the blockNode is sufficiently heavier than
+// 'cmp', where "sufficient" is defined as:
+//
+//     (1/bn.depth) - (1/cmp.depth) > (1/cmp.target) * SurpassThreshold
+func (bn *blockNode) heavierThan(cmp *blockNode) bool {
+	diff := new(big.Rat).Sub(bn.depth.Inverse(), cmp.depth.Inverse())
+	threshold := new(big.Rat).Mul(cmp.target.Inverse(), SurpassThreshold)
+	return diff.Cmp(threshold) > 0
 }
 
-// Rat returns a Target as a big.Rat.
-func (t Target) Rat() *big.Rat {
-	return new(big.Rat).SetInt(t.Int())
-}
+// timestampSlice is used for sorting.
+type timestampSlice []Timestamp
+
+func (ts timestampSlice) Len() int           { return len(ts) }
+func (ts timestampSlice) Less(i, j int) bool { return ts[i] < ts[j] }
+func (ts timestampSlice) Swap(i, j int)      { ts[i], ts[j] = ts[j], ts[i] }
```

Comment:
yeah, I never got around to finding a good place for this. This file is the only place it's used, so top of the file is probably best.

I will note that the stdlib uses this style internally: http://golang.org/src/sort/sort.go?s=5371:5390#L223